src/main/java/net/helpscout/api/HTTPMethodWrapper.java

```diff
@@ -0,0 +1,70 @@
+package net.helpscout.api;
+
+/**
+ * @Author: ivan
+ * Date: 22.08.16
+ * Time: 23:19
+ */
+public class HTTPMethodWrapper {
+
+    private final static String METHOD_GET = "GET";
+    private final static String METHOD_POST = "POST";
+    private final static String METHOD_PUT = "PUT";
+    private final static String METHOD_DELETE = "DELETE";
+
+    private String apiKey;
+    private String baseUrl;
+
+    public HTTPMethodWrapper() {
+    }
+
+    public void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public void setBaseUrl(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    public <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
+
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody)) {
```

Comment:
All the methods follow the same `try/catch/catch` format, is there a way to extract this functionality?